core/src/main/scala/io/finch/endpoint/param.scala

```diff
@@ -70,15 +69,14 @@ abstract private[finch] class Params[F[_], G[_], A](name: String)(implicit
             case None =>
               F.pure(Output.payload(present(decoded.map(_.right.get))))
             case Some(es) =>
-              F.raiseError(Errors(es.map(t => Error.NotParsed(items.ParamItem(name), tag, t))))
+              F.raiseError(Errors(es.map(t => Error.ParamNotParsed(name, tag).initCause(t).asInstanceOf[Error])))
```

Comment:
Why do you need `asInstanceOf` here?

Reply:
`initCause` gives back Throwable and it trips NonEmptyList.map.